tests/schedulers/airflow/test_dag_generator.py

```diff
@@ -0,0 +1,124 @@
+import typing as t
+from pytest_mock.plugin import MockerFixture
+
+from sqlglot import parse_one
+from airflow.models import BaseOperator
+from airflow.utils.context import Context
+
+from sqlmesh.core.config import EnvironmentSuffixTarget
+from sqlmesh.core.environment import Environment
+from sqlmesh.core.model import IncrementalByUniqueKeyKind, SqlModel
+from sqlmesh.core.snapshot import (
+    Snapshot,
+    SnapshotChangeCategory,
+)
+from sqlmesh.schedulers.airflow.dag_generator import SnapshotDagGenerator
+from sqlmesh.schedulers.airflow import common
+from sqlmesh.schedulers.airflow.operators.targets import BaseTarget, SnapshotEvaluationTarget
+from sqlmesh.utils.date import to_datetime
+
+
+class TestSubmitOperator(BaseOperator):
+    def __init__(
+        self,
+        *,
+        target: BaseTarget,
+        **kwargs: t.Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.target = target
+
+
+def test_generate_plan_application_dag__batch_index_populated(mocker: MockerFixture, make_snapshot):
```

Comment:
I couldn't find any existing tests for `SnapshotDagGenerator` so I added this file